app/controllers/harvests_controller.rb

```diff
@@ -90,7 +99,14 @@ def destroy
   private
 
   def harvest_params
-    params.require(:harvest).permit(:crop_id, :harvested_at, :description, :owner_id,
+    params.require(:harvest).permit(:planting_id, :crop_id, :harvested_at, :description, :owner_id,
       :quantity, :unit, :weight_quantity, :weight_unit, :plant_part_id, :slug, :si_weight)
   end
+
+  def planting_is_mine!
+    return unless @harvest.planting_id
+    @planting = Planting.find(@harvest.planting_id)
+    raise "Planting owner does not match" unless @planting.owner_id == current_member.id
```

Comment:
So we model this with
```
      can :create,  Planting
      can :update,  Planting, garden: { owner_id: member.id }
      can :destroy, Planting, garden: { owner_id: member.id }
```

It should be checked by something like
```can? :update @planting``` or ```authorise! :update @planting```
... though for some reason we have only really done this kind of checking at the view layer, and rarely in the controller layer.

Its probably because it happens implicitly in a lot of places:
> ... you can use the load_and_authorize_resource method in your controller to load the resource into an instance variable and authorize it automatically for every action in that controller.